parties/opkclient.go

```diff
@@ -21,55 +24,87 @@ type MFACos interface {
 }
 
 type OpkClient struct {
-	PktJson     []byte
-	Signer      *pktoken.Signer
+	Pkt         *pktoken.PKToken
 	Op          OpenIdProvider
 	MFACosigner MFACos
 }
 
-func (o *OpkClient) OidcAuth() ([]byte, error) {
-	nonce, err := o.Signer.GetNonce()
+func (o *OpkClient) OidcAuth(
+	signer crypto.Signer,
+	alg jwa.KeyAlgorithm,
```

Comment:
could we put the alg inside the signer? Or is there something special about `jws` algorithms that the crypto.Signer doesn't support?

If we can't combine them can we add a sanity check so someone doesn't accidentally put an alg that doesn't match the crypto.Signer's algorithm?

Reply:
Hmmm, I see the resolved comment below. Is there a jwk signer object that we could use?

Reply:
Johnny pointed out the exact same https://github.com/openpubkey/openpubkey/pull/39#discussion_r1367172012, I explained my reasoning here and tagged you in the conversation. He originally resolved it, but I unresolved so you could see.